features/netsocket/CellularInterface.h

```diff
@@ -23,6 +23,8 @@
  *  Common interface that is shared between Cellular interfaces
```

Comment:
What is the difference between mbed-os/features/unsupported/net/cellular/CellularModem/CellularModem.h and this interface? I see couple of APIs like connect/disconnect that are common. Why is this new interface missing ATCommandsInterface or SMS APIs?